docs/reference/core/ambassador.md

```diff
@@ -35,6 +35,7 @@ spec:
 | `envoy_log_format` | Defines the envoy log line format. See [this page](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log) for a complete list of operators | See [this page](https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log#default-format-string) for the standard log format. |
 | `envoy_log_path` | Defines the path of log envoy will use. By default this is standard output | `envoy_log_path: /dev/fd/1` |
 | `envoy_log_type` | Defines the type of log envoy will use, currently only support json or text | `envoy_log_type: text` |
+| `idle_timeout` | Controls how Envoy configures the tcp idle timeout on the http listener. Default is no timeout (TCP connection may remain idle indefinitely). NOTE: only s (seconds) is supported for the duration suffix | `idle_timeout: 300s` |
```

Comment:
Everywhere else, we do timeouts like `idle_timeout_ms` with an integer value in milliseconds. I kinda hate to say it, but let's change to that here as well?

Reply:
Thanks for the feedback.

I'll take a look at changing the value to accept an integer rather than a duration.

Regarding the name change from `idle_timeout`... If we change it to `idle_timeout_ms`, then this directly collides with the [idle_timeout_ms](https://www.getambassador.io/reference/timeouts/#idle-timeout-idle_timeout_ms) field on the upstream connection side.

Any suggestions or proposals on what to name it so that we can land somewhere close to the envoy naming, but still have it distinct from the existing attribute. Technically, this property is on the http listener, so maybe something like `listener_idle_timeout_ms` ? Are there other examples of similarly-named attributes that have to exist both on the upstream and the downstream?

Reply:
Oh, crap, I missed that collision. 😐 

We do sometimes use the names of Envoy things to differentiate, yes. I'd vote for probably `client_idle_timeout_ms`? but `listener_...` is OK with me too.